src/main/kotlin/calculator/Parser.kt

```diff
@@ -0,0 +1,22 @@
+package calculator
+
+import calculator.dto.Token
+import calculator.dto.Tokens
+
+// TODO-review : 상태를 갖지 않은 객체이기에 object 로 Parser 를 선언했습니다.
+object Parser {
+
+    fun parse(text: String): Tokens {
+        val result = StringAddCalculator.PARSER_RULE.find(text)
+        val delimiter = result?.groupValues?.get(1)
+
+        val tokens = when (delimiter) {
+            null -> text.split(StringAddCalculator.STD_DELIMITER)
+            else -> result.groupValues[2].split(delimiter)
+        }
```

Comment:
> 11번 라인을 돌려봐야 standard delimiter 를 사용할 수 있는지에 대한 확인이 가능한 로직입니다.
혹시 분기문을 작성하지 않고 Parser 를 구현할 수 있는 방법이 있을까요?

분기문 작성 없이는 어렵지 않을까요? 🤔
정규식으로 찾은 그룹들이 존재할지 아닐지를 체크하지 않고서는 다음 로직을 수행할 수 없을 것 같아요. 👀

if 식을 이용해서 조금 더 깔끔하게 다듬어볼 수 있을 것 같습니다. 
(delimiter 변수를 불필요한 타이밍에는 선언하지 않도록 변경해볼 수 있을 것 같아요. 🤔)

```suggestion
        val tokens = if (result != null) {
            val customDelimiter = result.groupValues[1]
            result.groupValues[2].split(customDelimiter)
        } else {
            text.split(StringAddCalculator.STD_DELIMITER)
        }
```